### PR TITLE
Seasons

### PR DESCRIPTION
**Pull request by owocki**
# What

Seasons

Implements https://github.com/owocki/slack_gamebot/issues/3
# Img

# Review

@bminortx

### CODE REVIEW

File: history/management/commands/run_bot.py [hunk 1]
```diff
@@ -155,26 +160,54 @@ def version(message):
...
+            if seasoned:
+                range_start_date = active_season.start_on
+            else:
+                active_season = None
+
+            return active_season, range_start_date
+
+        @listen_to('^gamebot alltime leaderboard (.*)',re.IGNORECASE)
+        @listen_to('^gb alltime leaderboard (.*)', re.IGNORECASE)
+        @listen_to('^alltime leaderboard (.*)',re.IGNORECASE)
```
Comment (bminortx) [inline history/management/commands/run_bot.py:181]: I do think we should move towards the `gamebot` or `gb` prefix for all commands eventually. It's a habit-breaker for the office, but I think it makes more sense for gamebot in a general slackbot sense.

Neither here nor there for this PR, though.

File: history/management/commands/run_bot.py [hunk 2]
```diff
@@ -155,26 +160,54 @@ def version(message):
...
         @listen_to('^gamebot leaderboard (.*)',re.IGNORECASE)
         @listen_to('^gb leaderboard (.*)', re.IGNORECASE)
         @listen_to('^leaderboard (.*)',re.IGNORECASE)
-        def leaderboard(message,gamename):
+        def seasoned_leaderboard(message,gamename):
+            return _leaderboard(message,gamename,True)
+
+        def _leaderboard(message,gamename,seasoned=False):
             #input sanitization
             gamename = gamename.strip()
```
Comment (bminortx) [inline history/management/commands/run_bot.py]: Sanitization should make it lowercase as well. Could this be a quick change in this PR? That way we could avoid recording Chess and chess and chesS. I'm pretty sure this isn't implemented yet, though I could be wrong.

File: history/management/commands/run_bot.py [hunk 3]
```diff
@@ -185,10 +218,47 @@ def leaderboard(message,gamename):
...
+            active_season, start_on = get_active_season(gamename,True)
+
+            #msg back to users
+            msg_str = "{} is active. \nUse `gamebot end season {}` to end this season.".format(active_season, gamename)
+            message.send(msg_str)
+
+
+        @listen_to('^end season (.*)',re.IGNORECASE)
+        @listen_to('^gamebot end season (.*)',re.IGNORECASE)
+        @listen_to('^gb end season (.*)',re.IGNORECASE)
```
Comment (bminortx) [inline history/management/commands/run_bot.py:243]: I feel like we should have some sort of lock on this command. I could see it being abused pretty easily. Any ideas on that front? Not necessary?

File: history/management/commands/run_bot.py [hunk 4]
```diff
@@ -340,11 +436,12 @@ def won(*arg):
             #setup
             sender = "@" + message.channel._client.users[message.body['user']][u'name']
             opponentname = _get_user_username(message,opponentname)
+            active_season , range_start_date = get_active_season(gamename,True)
 
             winner_old_elo = 1000
             loser_old_elo = 1000
-            if gamename == "chess":
-                elo_rankings = _get_elo(gamename)
+            if gamename in settings.ELO_ENABLED_GAMES:
```
Comment (bminortx) [inline history/management/commands/run_bot.py:443]: This is pretty.

File: history/management/commands/run_bot.py (hunk 1 above)
Comment (owocki) [inline history/management/commands/run_bot.py:181]: (whiney voice) buttttt i like having to type less...

File: history/management/commands/run_bot.py (hunk 2 above)
Comment (owocki) [inline history/management/commands/run_bot.py]: ya. will add that now

File: history/management/commands/run_bot.py (hunk 3 above)
Comment (owocki) [inline history/management/commands/run_bot.py:243]: i kind of feel 'meh' about it.  you and i have django admin access if we need to revert anything someone did.

File: history/management/commands/run_bot.py (hunk 2 above)
Comment (owocki) [inline history/management/commands/run_bot.py]: 3476f3e7856b808da25f179106b9c4907efca4a6

**Comment by owocki** [comment]
@bminortx ok to :ship: ?

**Comment by bminortx** [comment]
Make it so.

**Pull request merged**